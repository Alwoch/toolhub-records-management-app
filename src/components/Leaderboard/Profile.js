const Profile = ({id,img,fullName,score}) => {
  return (
    <>
      <div className="flex">
        <div className="item">
          <img src={img} alt="contributor profile" />
            <div className="info">
              <h3 className="name text-dark">{fullName}</h3>
            </div>
        </div>
        <div className="item">
          <span>{score}</span>
        </div>
      </div>;
    </>
  );
};

export default Profile;
