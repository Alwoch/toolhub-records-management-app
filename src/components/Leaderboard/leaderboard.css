Table{
 width:80%;
 margin: 1rem 0;
 border-bottom:2px solid var(--grey-600);
}
 .brief{
 color:var(--grey-500)
}
th{
 text-align: start;
}

tr th{
 border-bottom: 2px solid var(--grey-600)
}

td{
 padding:0 0.5rem;
 cursor: pointer;
}
tbody tr:hover{
 background-color: var(--grey-300);
}

@media only screen and (max-width:400px) {
 Table{
  width:100%
 }
}