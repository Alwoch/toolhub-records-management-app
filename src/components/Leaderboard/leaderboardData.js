const leaderboardData = [
  {
    id: 1,
    fullName: "Alwoch Sophia",
    score: "700",
    username:"@salwoch"
  },
  {
    id: 2,
    fullName: "Cassandra Lynn",
    score: "400",
    username:"@cassie",
  },
  {
    id: 3,
    fullName: "Alum Janat Massy",
    score: "370",
    username:"@massy",
  },
  {
    id: 4,
    fullName: "Atim Norah",
    score: "280",
    username:"@norie"
  },
  {
    id: 5,
    fullName: "Jim Sammy",
    score: "200",
    username:"@sam"
  },
  {
    id: 6,
    fullName: "Naggayi Daphne",
    score: "600",
    username:"@pearl"
  },
];

export default leaderboardData;
