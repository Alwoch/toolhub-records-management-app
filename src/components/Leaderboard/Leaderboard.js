import Navbar from "../Nav/Navbar";
import "./leaderboard.css";
import Profile from "./Profile";

const Leaderboard = () => {
  return (
    <>
    <Navbar/>
    <section className=".container-xxl" data-testid="leaderboard">
      <h1 className="leaderboard">Leaderboard</h1>
      <p className="brief">These users helped improve toolhub records in the last 30 days!</p>
      <Profile/>
    </section>
    </>
  );
};

export default Leaderboard;
