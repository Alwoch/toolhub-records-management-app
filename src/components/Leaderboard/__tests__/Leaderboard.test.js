import { render,screen} from '@testing-library/react';
import renderer from "react-test-renderer"
import { BrowserRouter } from "react-router-dom";
import Leaderboard from '../Leaderboard';

//Tests cases
  //Leaderboard should render correctly
  //Leaderboard should match the snapshot

test("should render Navbar component successfully",()=>{
 render(<BrowserRouter><Leaderboard/></BrowserRouter>);
 const leaderboardComponent=screen.getByTestId('leaderboard');
 expect(leaderboardComponent).toBeInTheDocument();
})

test("Navbar should match snapshot",()=>{
 const leaderboardTree=renderer.create(<BrowserRouter><Leaderboard/></BrowserRouter>).toJSON()
 expect(leaderboardTree).toMatchSnapshot()
})