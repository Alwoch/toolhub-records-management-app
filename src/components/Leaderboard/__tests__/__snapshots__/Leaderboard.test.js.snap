// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should match snapshot 1`] = `
Array [
  <nav
    className=".container-fluid"
    data-testid="nav"
  >
    <h3>
      Toolhub Records
    </h3>
    <div>
      <ul
        className="#nav-list"
        id="nav-list"
      >
        <li>
          <a
            className="active"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li>
          <a
            href="/dashboard"
            onClick={[Function]}
          >
            Dashboard
          </a>
        </li>
        <li>
          <a
            href="/leaderboard"
            onClick={[Function]}
          >
            Leaderboard
          </a>
        </li>
      </ul>
    </div>
    <div
      id="mobile"
      onClick={[Function]}
    >
      <i
        className="fas fa-bars"
        id="bar"
      />
    </div>
  </nav>,
  <section
    className=".container-xxl"
    data-testid="leaderboard"
  >
    <h1
      className="leaderboard"
    >
      Leaderboard
    </h1>
    <p
      className="brief"
    >
      These users helped improve toolhub records in the last 30 days!
    </p>
    <table
      className="table table-striped table-bordered table-hover"
    >
      <thead>
        <tr>
          <th>
            #
          </th>
          <th>
            Name
          </th>
          <th>
            username
          </th>
          <th>
            Score
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </section>,
]
`;
